### PR TITLE
Documentation for the OAuth2 integration

### PR DESCRIPTION
**Pull request by imankurpatel000**
I am creating a new pull request to merge the documentation done by @iCodr8 in the original Pull Request #182 Since he doesn't seem to have the original pull request branch and this documentation is pending for a long time. This refers to issue #180.

I have also made changes as suggested by @dunglas in that original pull request. Can someone please review this and let me know if this looks correct?

### CODE REVIEW

**Comment by imankurpatel000** [comment]
Hello guys, any update on this?

**Comment by dunglas** [comment]
@chalasr would you mind to take a look if it looks good to you? I never used FOSOAuth :/

File: core/oauth2.md [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+Add the file from the [FOSUserBundle Integration](fosuser-bundle.md#creating-a-user-entity-with-serialization-groups).
+
+## Security Configuration
+
+```yaml
+# To get started with security, check out the documentation:
+# http://symfony.com/doc/current/book/security.html
+security:
+    encoders:
+        FOS\UserBundle\Model\UserInterface: bcrypt
```
Comment (dunglas) [inline core/oauth2.md:185]: The `auto` mode should be used isn't it? https://symfony.com/blog/new-in-symfony-4-3-native-password-encoder

File: core/oauth2.md [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+    type: "api_platform"
+
+fos_oauth_server_token:
+    resource: "@FOSOAuthServerBundle/Resources/config/routing/token.xml"
+
+fos_oauth_server_authorize:
+    resource: "@FOSOAuthServerBundle/Resources/config/routing/authorize.xml"
+```
+
+## Create clients
```
Comment (dunglas) [inline core/oauth2.md:236]:
```suggestion
## Create Clients
```

File: core/oauth2.md [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+
+Add this class to `AppBundle/Command/CreateOAuthClientCommand.php` to generate oauth clients via console:
```
Comment (dunglas) [inline core/oauth2.md:238]:
```suggestion
Add this class to `api/src/Command/CreateOAuthClientCommand.php` to generate Oauth clients via console:
```

Other paths and namespaces must be updated accordingly.

File: core/oauth2.md [hunk 4]
```diff
@@ -0,0 +1,329 @@
...
+    }
+}
+
+```
+
+```php
+<?php
+// AppBundle/Entity/RefreshToken.php
+
+namespace AppBundle\Entity;
```
Comment (dunglas) [inline core/oauth2.md:143]:
```suggestion
namespace App\Entity;
```

Same everywhere

File: core/oauth2.md [hunk 5]
```diff
@@ -0,0 +1,329 @@
...
+namespace AppBundle\Command;
+
+use AppBundle\Entity\Client;
+use Symfony\Bundle\FrameworkBundle\Command\ContainerAwareCommand;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+class CreateOAuthClientCommand extends ContainerAwareCommand
```
Comment (chalasr) [inline core/oauth2.md:252]: The bundle already provides [a similar command](https://github.com/FriendsOfSymfony/FOSOAuthServerBundle/blob/master/Command/CreateClientCommand.php) out of the box, isn't it?

**Comment by imankurpatel000** [comment]
@chalasr You are right, the FOSOAuthServerBunde is not ready for Symfony 5. I actually ended up using trikoder/oauth2-bundle instead of FOSOAuthServerBunde for OAuth in our project. I can try to write documentation for integrating trikoder/oauth2-bundle instead of FOSOAuthServerBunde with api platform. trikoder/oauth2-bundle is already compatible with Symfony 5. Let me know if that is something that would be useful.

**Comment by vinceAmstoutz** [comment]
A deprecated version of API Platform. Feel free to reopen it in our last version.

**Pull request closed without merging**